### PR TITLE
define IOContext(io, pairs...) and remove IOContext(io; kw...)

### PR DESCRIPTION
**Pull request by rfourquet**
* Usually a context is defined via a pair (`IOContext(io, :k=>v)`),
  but if another pair has to be pushed, one had to either add an
  IOContext call (`IOContext(IOContext(io, :k=>v), :x=>y)`) or to
  switch to kwargs syntax (`IOContext(io, k=v, x=y)`), none
  of which is satisfactory.
* As a consequence of allowing passing multiple pairs, the
  method using keyword arguments is obsoleted (and hence deleted).
* Also, the method `IOContext(io, key, value)` is deleted in
  favor of `IOContext(io, key=>value)`.

I originally only wanted to allow `IOContext(io, pairs...)`, but in the process saw opportunities for simplification of the API, as `IOContext(io, :key, value)` and `IOContext(io, key=value)` were almost not used. I have no problem to restore them if they are deemed necessary.

### CODE REVIEW

**Comment by rfourquet** [comment]
So a bit of history:
* `IOContext` is introduced in 4706184a424eda72aa802f465c7f45c5545143e0;
* the kewyord version was introduced in ce231746d0413fb6f3acc23c41abe4240837d858, apparently to construct more easily `IOContext` with more than 1 property;
* in 893a65bddea0bd89c3df2eb0cf8727b1f967738d, many keyword calls are converted into pair calls (including when there is more than one property! i.e. `IOContext(io, a=1, b=2)` transformed into `IOContext(IOContext(io, :a=>1), :b=>2)`), probably for better performance;
* in 8db51e618dd8adea9f19436753d75db001f7914e, some pair calls are converted into keyword calls (not sure why).

In other words, having two concurrent APIs is kind of messy; as the prefered syle seems to use pairs, and it's faster (as of today), this confirms my impression that we should get rid of the keyword version, which can I think accomplish nothing more than the pairs version. So I added a deprecation, and I will remove the WIP. 

As the main authors of `IOContext`, @vtjnash @JeffBezanson would you mind having a look?

File: base/show.jl [hunk 1]
```diff
@@ -20,41 +20,32 @@ struct IOContext{IO_t <: IO} <: AbstractPipe
     end
 end
 
-"""
-    IOContext(io::IO; properties...)
+unwrapcontext(io::IO) = io, ImmutableDict{Symbol,Any}()
+unwrapcontext(io::IOContext) = io.io, io.dict
```
Comment (rfourquet) [inline base/show.jl]: I introduced these two methods as they often allow to merge two methods (for `IO` and for `IOContext`) into one; moreover, they make it easy to spot what is discarded (e.g. in `IOContext(unwrapcontext(io)[1], unwrapcontext(context)[2])`, only the stream of `io` and the dict of `context` are kept).

**Comment by fredrikekre** [comment]
If you feel like fixing an issue while you are at it, there is #22637 to add some more examples :)

**Comment by rfourquet** [comment]
> If you feel like fixing an issue while you are at it, there is #22637 to add some more examples :)

OK, I can do that! but will use another PR.

**Comment by rfourquet** [comment]
`IOContext` is described as being "an immutable dictionary that is a subclass of `IO`". The API for a dictionary is to use pairs, so I think this PR brings more consistency. I will merge in a couple of days if no objection.

**Review by StefanKarpinski — APPROVED** [review]
I agree that this is much nicer and more consistent.

**Comment by rfourquet** [comment]
Wow, it became rare to see CI all green :)
